cursor.go

```diff
@@ -40,22 +40,24 @@ type CursorConfig struct {
 }
 
 // returns a new Cursor with an embedded copy of itself
-func (c *CursorConfig) NewCursor(table *schema.Table, maxPk uint64) *Cursor {
+func (c *CursorConfig) NewCursor(table *schema.Table, startPk, maxPk uint64) *Cursor {
 	return &Cursor{
-		CursorConfig:  *c,
-		Table:         table,
-		MaxPrimaryKey: maxPk,
-		RowLock:       true,
+		CursorConfig:             *c,
+		Table:                    table,
+		MaxPrimaryKey:            maxPk,
+		RowLock:                  true,
+		lastSuccessfulPrimaryKey: startPk,
 	}
 }
 
 // returns a new Cursor with an embedded copy of itself
-func (c *CursorConfig) NewCursorWithoutRowLock(table *schema.Table, maxPk uint64) *Cursor {
+func (c *CursorConfig) NewCursorWithoutRowLock(table *schema.Table, startPk, maxPk uint64) *Cursor {
 	return &Cursor{
```

Comment:
```golang
cursor := c.NewCursor(table, startPk, maxPk)
cursor.RowLock = false
return cursor
```
seems simpler maybe and more in line with what you are doing below with the `WithoutStateTracker` method?